src/components/GoodsList.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+
+import { get5First, getAll, getRedGoods } from '../api/goods';
+
+type State = {
+  colors: Good[];
+};
+
+type Props = {
+  type: 'fullColors' | 'firstFive' | 'redColors' | '';
+};
+
+export class GoodsList extends React.Component<Props, State> {
+  state: State = {
+    colors: [],
+  };
+
+  async componentDidUpdate(prevProps: Props) {
+    if (prevProps.type !== this.props.type) {
+      const newColors = await this.pritNewColor();
+
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ colors: [...newColors] });
+    }
+  }
```

Comment:
why do you need componentDidUpdate here? you should call a function when one of the button pressed. You no need lifecycle method in this case